### PR TITLE
Add documentation and example for write-module-hook

### PR DESCRIPTION
**Pull request by Flamefire**
Also fix some typos and minor issues

### CODE REVIEW

**Review by boegel — APPROVED** [review]

File: docs/Hooks.rst [hunk 1]
```diff
@@ -119,6 +123,12 @@ Do take into account the following:
 * for ``parse_hook``, one argument is provided: the ``EasyConfig`` instance
   that corresponds to the easyconfig file being parsed (usually referred to as ``ec``)
 
+* for ``module_write_hook``, 3 arguments are provided:
+   * the ``EasyBlock`` instance used to perform the installation (usually referred to as ``self``)
+   * the filepath of the module that will be written
+   * the module text as a string
+  The return value of this hook (if any) will then be appended to the contents of the module file.
```
Comment (Flamefire) [inline docs/Hooks.rst:130]: @boegel You changed this sentence and introduced a mistake: The contents will be **written** to the module file. NOT appended, just as I wrote initially. The whole return value will be used and written directly, so this is misleading

Or what did you mean by "appended" here?

Comment (boegel) [reply docs/Hooks.rst:130]: Oh, and I even checked the code before changing this... Will fix.

Comment (Flamefire) [reply docs/Hooks.rst:130]: For rerference see https://github.com/easybuilders/easybuild-framework/blob/96977045496d8a5106a54727f6522c92863008e8/easybuild/framework/easyblock.py#L3168-L3170 and  https://github.com/easybuilders/easybuild-framework/blob/96977045496d8a5106a54727f6522c92863008e8/easybuild/framework/easyblock.py#L3179

Comment (boegel) [reply docs/Hooks.rst:130]: Fixed in #715, thanks for double checking and catching this!

**Pull request merged**